src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,85 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import cn from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { getVisibleTodos } from '../../helpers/getVisibleTodos';
+
+export const TodoList: React.FC = React.memo(() => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const selectTodo = (todo: Todo) => {
+    dispatch(currentTodoActions.setTodo(todo));
+  };
+
+  const visibleTodos = useMemo(() => (
+    getVisibleTodos(todos, query, status)
+  ), [todos, query, status]);
```

Comment:
It could be custom hook that don't take arguments and returns `visibleTodos`
```
const useVisibleTodos = () => {
  // call selectors

 // return useMemo resuls
}